src/context/service-worker-context.tsx

```diff
@@ -9,8 +25,32 @@ export const ServiceWorkerContext = createContext({
 export const ServiceWorkerProvider = ({ children }): JSX.Element => {
   const [isServiceWorkerRegistered, setIsServiceWorkerRegistered] = useState(false)
 
+  const windowLocation = translateIpfsRedirectUrl(window.location.href)
+
   useEffect(() => {
     if (isServiceWorkerRegistered) {
+      /**
+       * The service worker is registered, now we need to check for "helia-sw" and origin isolation support
+       */
+      if (windowLocation.href !== window.location.href) {
+      /**
+       * We're at a domain with ?helia-sw=, we can reload the page so the service worker will
+       * capture the request
+       */
+        window.location.replace(windowLocation.href)
+      } else {
+        /**
+         * ?helia-sw= url handling is done, now we can check for origin isolation redirects
+         */
+        void findOriginIsolationRedirect(windowLocation).then((originRedirect) => {
+          if (originRedirect !== null) {
+            window.location.replace(originRedirect)
+          }
+        })
```

Comment:
`/?helia-sw=` redirect will load path URL.. if not handled by service worker for whatever reason (it should be, 99.99% of the time) then this will redirect that path request to subdomain if supported.